ads/google/a4a/utils.js

```diff
@@ -286,7 +286,11 @@ export function googlePageParameters(a4a, startTime) {
       dev().expectedError('AMP-A4A', 'Referrer timeout!');
       return '';
     });
-  const domLoading = getNavigationTiming(win, 'domLoading');
+  // Set dom loading time to first visible if page started in prerender state
+  // determined by truthy value for visibilityState param.
+  const domLoading = a4a.getAmpDoc().getParam('visibilityState') ?
```

Comment:
IIUC, this is not equivalent to `AmpDoc#isVisible`. Is there a more direct accessor? (In the spirit of https://testing.googleblog.com/2017/07/code-health-to-comment-or-not-to-comment.html)

Reply:
Ad requests are not sent when in a prerender state.  Instead they delay until the page becomes visible so asking if isVisible would always return true.  What we need is whether it started in a prerender state and this appears to be the only way to do that.  If page does not start in a prerender state, getLastVisible === time AMP runtime started which is also not what we want here.